### PR TITLE
Feature/gh 9774

### LINKED ISSUE #9774: Disable 'Disable' button while we have Edit form open
Disabling item while have edit form open should be prohibited.

Check on:
- [x] budget details
- [x] cost center details
- [x] unit details
- [x] user details
- [x] permission details

I think enable button should be still avaliable (if is not blocked by other reason) for case if someone open form from url, then he can unblock item and fill the form

### PR DESCRIPTION
**Pull request by mszczepaniak**
Closes #9774

### CODE REVIEW

File: feature-libs/organization/administration/components/shared/detail/toggle-status-action/toggle-status.component.spec.ts [hunk 1]
```diff
@@ -24,6 +24,8 @@ class MockMessageService {
 
 class MockItemService {
   current$ = of();
+  isInEditMode$: Observable<boolean> = new BehaviorSubject<boolean>(false);
```
Comment (dunqan) [inline feature-libs/organization/administration/components/shared/detail/toggle-status-action/toggle-status.component.spec.ts:27]: maybe we should test if this behavior is correct?

File: feature-libs/organization/administration/components/shared/form/form.component.ts [hunk 2]
```diff
@@ -81,9 +88,21 @@ export class FormComponent<T> {
     this.i18n = this.i18nRoot + (item ? '.edit' : '.create');
   }
 
+  protected setIsInEdit(): void {
```
Comment (dunqan) [inline feature-libs/organization/administration/components/shared/form/form.component.ts]: the logic is so compact now, what about dropping this method and moving it to ngOnInit directly?

File: feature-libs/organization/administration/components/shared/item.service.ts [hunk 3]
```diff
@@ -109,4 +111,10 @@ export abstract class ItemService<T> {
   getRouterParam(key: string): Observable<string> {
     return this.currentItemService.getRouterParam(key);
   }
+  /**
+   * Sets to true when the user is on the entity item form page
+   */
+  setEditMode(isInEdit: boolean) {
```
Comment (dunqan) [inline feature-libs/organization/administration/components/shared/item.service.ts:118]: Shoudln't we cover this with unit tests?

File: feature-libs/organization/administration/components/shared/form/form.component.ts (hunk 2 above)
Comment (mszczepaniak) [reply feature-libs/organization/administration/components/shared/form/form.component.ts]: agree

File: feature-libs/organization/administration/components/shared/item.service.ts (hunk 3 above)
Comment (mszczepaniak) [reply feature-libs/organization/administration/components/shared/item.service.ts:118]: yes indeed

File: feature-libs/organization/administration/components/shared/detail/toggle-status-action/toggle-status.component.spec.ts (hunk 1 above)
Comment (mszczepaniak) [reply feature-libs/organization/administration/components/shared/detail/toggle-status-action/toggle-status.component.spec.ts:27]: yes indeed

File: feature-libs/organization/administration/components/shared/form/form.component.spec.ts [hunk 4]
```diff
@@ -22,6 +22,8 @@ class MockItemService {
   save() {
     return of({ status: LoadStatus.SUCCESS, item: mockItem });
   }
+  load = createSpy('load').and.returnValue(of());
+  setEditMode = () => false;
```
Comment (dunqan) [inline feature-libs/organization/administration/components/shared/form/form.component.spec.ts]: should be `() => {}` to be less confusing

File: feature-libs/organization/administration/components/shared/item.service.spec.ts [hunk 5]
```diff
@@ -150,5 +150,29 @@ describe('ItemService', () => {
         expect(form.disabled).toBeFalsy();
       });
     });
+
+    describe('isInEditMode', () => {
+      let result;
```
Comment (dunqan) [inline feature-libs/organization/administration/components/shared/item.service.spec.ts]: you should clear or reset the result in `beforeEach`, or move this `let` inside `it`

**Review by dunqan — APPROVED** [review]

**Pull request merged**